Sources/TuistKit/Generator/ModelLoader.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+import Basic
+
+/// Entity responsible for providing generator models
+///
+/// Assumptions:
+///   - TuistGenerator creates a graph of Project dependencies
+///   - The projects are associated with unique paths
+///   - Each path only contains one Project
+///   - Whenever a dependency is encountered referencing another path,
+///     this entity is consulted again to load the model at that path
+protocol ModelLoading {
```

Comment:
I'd call this one `GeneratorModelLoading`

Reply:
Sure will push the changes